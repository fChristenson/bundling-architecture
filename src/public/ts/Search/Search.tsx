import React from "react";

interface IProps {
  setNumber(n: number): void;
}

export const Search: React.FunctionComponent<IProps> = ({
  children,
  setNumber,
}) => {
  return (
    <div>
      <input
        type="text"
        onChange={(e) => setNumber(parseInt(e.target.value))}
        placeholder="Filter"
      />
      {children}
    </div>
  );
};
