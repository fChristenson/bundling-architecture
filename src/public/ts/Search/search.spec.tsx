import { Search } from "./Search";
import React from "react";
import render from "react-test-renderer";

const noop = () => {};

describe("Search", () => {
  it("should render", () => {
    const search = render.create(<Search setNumber={noop}>foo</Search>);
    const node = search.toJSON();
    expect(node).toMatchSnapshot();
  });
});
