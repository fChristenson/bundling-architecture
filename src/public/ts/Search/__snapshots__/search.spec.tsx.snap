// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search should render 1`] = `
<div>
  <input
    onChange={[Function]}
    placeholder="Filter"
    type="text"
  />
  foo
</div>
`;
